MSAL/src/instance/MSALAADAuthority.m

```diff
@@ -51,6 +52,102 @@ - (nullable instancetype)initWithURL:(nonnull NSURL *)url
     return self;
 }
 
+- (instancetype)initWithCloudInstance:(MSALAzureCloudInstance)cloudInstance
+                         audienceType:(MSALAudienceType)audienceType
+                            rawTenant:(NSString *)rawTenant
+                                error:(NSError **)error
+{
+    NSString *environment = [self environmentFromCloudInstance:cloudInstance];
+    
+    if (!environment)
+    {
+        if (error)
+        {
+            NSError *msidError = MSIDCreateError(MSIDErrorDomain, MSIDErrorInvalidDeveloperParameter, @"Invalid MSALAzureCloudInstance provided", nil, nil, nil, nil, nil);
+            *error = [MSALErrorConverter msalErrorFromMsidError:msidError];
+        }
+        
+        return nil;
+    }
+    
+    if (![NSString msidIsStringNilOrBlank:rawTenant])
+    {
+        if (audienceType != MSALAzureADMyOrgOnlyAudience)
+        {
+            if (error)
+            {
+                NSError *msidError = MSIDCreateError(MSIDErrorDomain, MSIDErrorInvalidDeveloperParameter, @"Invalid MSALAudienceType provided. You can only provide rawTenant when using MSALAzureADMyOrgOnlyAudience.", nil, nil, nil, nil, nil);
+                *error = [MSALErrorConverter msalErrorFromMsidError:msidError];
+            }
+            
+            return nil;
+        }
+        
+        NSURL *aadURL = [NSURL msidURLWithEnvironment:environment tenant:rawTenant];
+        return [self initWithURL:aadURL rawTenant:nil error:error];
+    }
+    
+    NSString *audienceString = [self audienceFromType:audienceType];
+    
+    if (!audienceString)
+    {
+        if (error)
+        {
+            NSError *msidError = MSIDCreateError(MSIDErrorDomain, MSIDErrorInvalidDeveloperParameter, @"Invalid MSALAudienceType provided. You must provide rawTenant when using MSALAzureADMyOrgOnlyAudience.", nil, nil, nil, nil, nil);
```

Comment:
nit: Should this error be generated inside `audienceFromType:`? If we introduce a new enum type that converts to 'nil' string, this error messages will be incorrect.